### PR TITLE
HtmlEditor - extract a base module

### PR DESCRIPTION
**Pull request by alexander-kotov-dx**

### CODE REVIEW

File: js/ui/html_editor/modules/resizing.js [hunk 1]
```diff
@@ -19,14 +19,17 @@ const MOUSEDOWN_EVENT = addNamespace('mousedown', MODULE_NAMESPACE);
...
     constructor(quill, options) {
         this.quill = quill;
         this.editorInstance = options.editorInstance;
+
         this.allowedTargets = options.allowedTargets || ['image'];
         this.enabled = !!options.enabled;
         this._hideFrameWithContext = this.hideFrame.bind(this);
         this._framePositionChangedHandler = this._prepareFramePositionChangedHandler();
 
+
```
Comment (DokaRus) [inline js/ui/html_editor/modules/resizing.js]:
```suggestion
```

File: js/ui/html_editor/modules/dropImage.js [hunk 2]
```diff
@@ -6,16 +6,15 @@ import { each } from '../../../core/utils/iterator';
 import browser from '../../../core/utils/browser';
 import { getWindow } from '../../../core/utils/window';
 
+import BaseHtmlEditorModule from './base';
```
Comment (DokaRus) [inline js/ui/html_editor/modules/dropImage.js]:
```suggestion
import BaseModule from './base';
```
Keep it simple

Comment (ksercs) [inline js/ui/html_editor/modules/dropImage.js]: Seems like we shouldn't import this module where there is no Quill (the same is in popup and toolbar)

File: js/ui/html_editor/modules/base.js [hunk 3]
```diff
@@ -0,0 +1,25 @@
+import Quill from 'devextreme-quill';
+
+let BaseHtmlEditorModule = class EmptyModule {};
```
Comment (DokaRus) [inline js/ui/html_editor/modules/base.js]: Extract duplicated class

```suggestion
import EmptyModule from './empty';

let BaseHtmlEditorModule = EmptyModule;
```
And so on for other modules

File: js/ui/html_editor/modules/dropImage.js (hunk 2 above)
Comment (DokaRus) [reply js/ui/html_editor/modules/dropImage.js]: >The standard import syntax is static and will always result in all code in the imported module being evaluated at load time

File: js/ui/html_editor/modules/dropImage.js [hunk 4]
```diff
@@ -6,16 +6,16 @@ import { each } from '../../../core/utils/iterator';
 import browser from '../../../core/utils/browser';
 import { getWindow } from '../../../core/utils/window';
 
-let DropImageModule = {};
+import EmptyModule from './empty';
```
Comment (DokaRus) [inline js/ui/html_editor/modules/dropImage.js]:
```suggestion
```

File: js/ui/html_editor/modules/dropImage.js [hunk 5]
```diff
@@ -6,16 +6,16 @@ import { each } from '../../../core/utils/iterator';
...
+import BaseModule from './base';
 
-if(Quill) {
-    const BaseModule = Quill.import('core/module');
+let DropImageModule = EmptyModule;
```
Comment (DokaRus) [inline js/ui/html_editor/modules/dropImage.js]:
```suggestion
let DropImageModule = BaseModule;
```
If `!Quill` => `BaseModule === EmptyModule`

And so on for other modules

**Review by DokaRus — APPROVED** [review]

**Review by Zedwag — APPROVED** [review]

**Pull request merged**